std/range/package.d

```diff
@@ -4756,21 +4755,25 @@ body
                                             cast(Value)(pastLast - (length - upper) * step),
                                             step);
         }
-        @property IndexType length() const
+        @property size_t length() const
         {
             if (step > 0)
             {
-                return unsigned((pastLast - current) / step);
+                return cast(size_t)unsigned((pastLast - current) / step);
             }
             else
             {
-                return unsigned((current - pastLast) / -step);
+                return cast(size_t)unsigned((current - pastLast) / -step);
             }
         }
 
         alias opDollar = length;
     }
 
+    if (step > 0)
+        assert(unsigned((end - begin) / step) <= size_t.max);
+    else
+        assert(unsigned((end - begin) / -step) <= size_t.max);
```

Comment:
Probably only need this in 32-bit land. I think in 64-bit land, this is a noop :)